tasks/us/census/tiger.py

```diff
@@ -651,34 +660,48 @@ def requires(self):
         }
 
     def columns(self):
-        return OrderedDict([
+        cols = OrderedDict([
             ('geoid', self.input()['geoids'][self.geography + '_geoid']),
             ('the_geom', self.input()['geoms'][self.geography + '_clipped']),
             ('aland', self.input()['attributes']['aland']),
-            ('name', self.input()['geonames'][self.geography + '_geoname']),
         ])
+        if self.name():
+            cols['geoname'] = self.input()['geonames'][self.geography + '_geoname']
+        return cols
 
     def timespan(self):
         return self.year
 
     def populate(self):
         session = current_session()
 
-        stmt = ('''INSERT INTO {output}
-                   SELECT
-                     geoid,
-                     ST_Union(ARRAY(
-                       SELECT ST_MakePolygon(ST_ExteriorRing(
-                         (ST_Dump(the_geom)).geom
-                       ))
-                       WHERE GeometryType(the_geom) = 'POLYGON'
-                     )),
-                     aland
-                   FROM {input}'''.format(
-                    output=self.output().table,
-                    input=self.input()['data'].table), )[0]
+        additional_columns = ', geoname' if self.name() else ''
+        stmt = '''INSERT INTO {output}
+                  SELECT
+                    {input}.geoid,
+                    ST_Union(ARRAY(
+                      SELECT ST_MakePolygon(ST_ExteriorRing(
+                        (ST_Dump({input}.the_geom)).geom
+                      ))
+                      WHERE GeometryType({input}.the_geom) = 'POLYGON'
+                    )),
+                    {input}.aland
+                    {additional_columns}
+                  FROM {input}
+                  LEFT JOIN {original_table} USING (geoid)
+               '''.format(
+                   output=self.output().table,
+                   input=self.input()['data'].table,
+                   additional_columns=additional_columns,
+                   original_table=self.original_task().output().table)
         session.execute(stmt)
 
+    def original_task(self):
+        return self.requires()['data'].original_task()
+
+    def name(self):
+        return self.original_task().name
```

Comment:
I know this is one of the reasons to use the list of dependencies to get the "original task" but we could have a data structure or POPO which links name column and the geography level

Reply:
This is actually needed to get the original table name as well. If we remove that and go for propagating `name` through intermediate tables, I agree that it's easier to change this to a global configuration dict rather than going up the dependency chain.

Reply:
My concern is to couple our code with the internal functions of Luigi. Deal with dependencies and core things of Luigi, even if they're available, is kind of bad smell. We should isolate and minimize this kind of things.